packages/flutter_tools/lib/src/commands/analyze_once.dart

```diff
@@ -93,7 +94,9 @@ class AnalyzeOnce extends AnalyzeBase {
         arguments.add('--no-preview-dart-2');
       }
 
-      final String dartanalyzer = fs.path.join(Cache.flutterRoot, 'bin', 'cache', 'dart-sdk', 'bin', 'dartanalyzer');
+      final String sdkPath = argResults['dart-sdk'] ?? sdk.dartSdkPath;
```

Comment:
Does `sdk.dartSdkPath` correspond or default to `<flutter-root>/bin/cache/dart-sdk/bin/dartanalyzer` ?

Reply:
No, to `<flutter-root>/bin/cache/dart-sdk` - we join in to `bin/dartanalyzer` down below :)